crates/op-rbuilder/src/builders/flashblocks/payload.rs

```diff
@@ -1203,3 +1247,47 @@ where
         fb_payload,
     ))
 }
+
+/// Calculates only the state root for an existing payload
+fn calculate_state_root_only<DB, P, ExtraCtx>(
+    state: &mut State<DB>,
+    ctx: &OpPayloadBuilderCtx<ExtraCtx>,
+    info: &ExecutionInfo<FlashblocksExecutionInfo>,
+) -> Result<B256, PayloadBuilderError>
+where
+    DB: Database<Error = ProviderError> + AsRef<P>,
+    P: StateRootProvider + HashedPostStateProvider + StorageRootProvider,
+    ExtraCtx: std::fmt::Debug + Default,
+{
+    let state_root_start_time = Instant::now();
+    state.merge_transitions(BundleRetention::Reverts);
```

Comment:
you don't need to merge transitions as you don't have them, you should call this only after you send last bundle state